tests/testthat/test-implicit_assignment_linter.R

```diff
@@ -74,6 +74,26 @@ test_that("implicit_assignment_linter skips allowed usages", {
     NULL,
     linter
   )
+
+  expect_lint(
+    trim_some("
+      lapply(x, function(xi) {
+        xi <- xi + 1
+        xi
+      })"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      map(x, function(xi) {
```

Comment:
I had in mind calls with `~`, could you add one of those?

Reply:
Done. Lemme know if you have any other tests in mind.